DUI3-DX/Converters/Revit/Speckle.Converters.RevitShared/ToHost/Raw/Geometry/PolylineConverterToHost.cs

```diff
@@ -0,0 +1,96 @@
+using Speckle.Converters.Common.Objects;
+using Speckle.Converters.RevitShared.Helpers;
+using Speckle.Converters.RevitShared.Services;
+using Speckle.Core.Logging;
+
+namespace Speckle.Converters.RevitShared.ToSpeckle;
+
+public class PolylineConverterToHost : ITypedConverter<SOG.Polyline, DB.CurveArray>
+{
+  private readonly ITypedConverter<SOG.Line, DB.Line> _lineConverter;
+  private readonly ScalingServiceToHost _scalingService;
+  private readonly IRevitConversionContextStack _contextStack;
+
+  public PolylineConverterToHost(
+    ITypedConverter<SOG.Line, DB.Line> lineConverter,
+    ScalingServiceToHost scalingService,
+    IRevitConversionContextStack contextStack
+  )
+  {
+    _lineConverter = lineConverter;
+    _scalingService = scalingService;
+    _contextStack = contextStack;
+  }
+
+  public CurveArray Convert(Polyline target)
+  {
+    var curveArray = new CurveArray();
+    if (target.value.Count == 6)
+    {
+      // Polyline is actually a single line
+      TryAppendLineSafely(curveArray, new SOG.Line(target.value, target.units));
+    }
+    else
+    {
+      var pts = target.GetPoints();
+      var lastPt = pts[0];
+      for (var i = 1; i < pts.Count; i++)
+      {
+        var success = TryAppendLineSafely(curveArray, new SOG.Line(lastPt, pts[i], target.units));
+        if (success)
+        {
+          lastPt = pts[i];
+        }
+      }
+
+      if (target.closed)
+      {
+        TryAppendLineSafely(curveArray, new SOG.Line(pts[^1], pts[0], target.units));
+      }
+    }
+    return curveArray;
+  }
+
+  /// <summary>
+  /// Checks if a Speckle <see cref="SOG.Line"/> is too sort to be created in Revit.
+  /// </summary>
+  /// <remarks>
+  /// The length of the line will be computed on the spot to ensure it is accurate.
+  /// </remarks>
+  /// <param name="line">The <see cref="SOG.Line"/> to be tested.</param>
+  /// <returns>true if the line is too short, false otherwise.</returns>
+  public bool IsLineTooShort(SOG.Line line)
+  {
+    var scaleToNative = _scalingService.ScaleToNative(SOG.Point.Distance(line.start, line.end), line.units);
+    return scaleToNative < _contextStack.Current.Document.Application.ShortCurveTolerance;
+  }
+
+  /// <summary>
+  /// Attempts to append a Speckle <see cref="SOG.Line"/> onto a Revit <see cref="CurveArray"/>.
+  /// This method ensures the line is long enough to be supported.
+  /// It will also convert the line to Revit before appending it to the <see cref="CurveArray"/>.
+  /// </summary>
+  /// <param name="curveArray">The revit <see cref="CurveArray"/> to add the line to.</param>
+  /// <param name="line">The <see cref="SOG.Line"/> to be added.</param>
+  /// <returns>True if the line was added, false otherwise.</returns>
+  public bool TryAppendLineSafely(CurveArray curveArray, SOG.Line line)
+  {
+    if (IsLineTooShort(line))
+    {
+      // poc : logging "Some lines in the CurveArray where ignored due to being smaller than the allowed curve length."
```

Comment:
ahh hmmmm this is one of those grey areas. I presume we just brought this over but we should make some solid decision about whether this is valid as it's an effective mutation. Please discuss with @clairekuang and we can probably make a ticket and drop it here, even if that ticket is. Let's think about this later and either do something or close the ticket and remove the POC

Reply:
I guess Claire was away today but can you pick it afterwards?

Reply:
I'm not sure when this was first implemented, but clearly someone ran into the issue of trying to import a polyline with one segment that was so short that it couldn't be modeled in Revit. What we are doing here makes a lot of sense to me, just ignore the really small segment. The tolerance is .8mm so it seems unlikely that a user will even miss this segment of their polyline

Reply:
I can definitely talk to claire about it though